framework/moose.mk

```diff
@@ -317,6 +317,10 @@ $(moose_revision_header): $(moose_HEADER_deps)
 	@echo "Checking if header needs updating: "$@"..."
 	$(shell $(FRAMEWORK_DIR)/scripts/get_repo_revision.py $(FRAMEWORK_DIR) \
 	  $(moose_revision_header) MOOSE)
+        # make sure the header generation step didn't fail
+	@if [ $(.SHELLSTATUS) -ne 0 ]; then \
+	echo "\nFailed to generate MooseRevision.h\n"; exit $(.SHELLSTATUS); \
+	fi
 	@if [ ! -e "$(moose_all_header_dir)/MooseRevision.h" ]; then \
 		ln -sf $(moose_revision_header) $(moose_all_header_dir); \
```

Comment:
@loganharbour this code makes me think that we used to be fine with not having this header?